homeassistant/components/nasweb/commons.py

```diff
@@ -0,0 +1,46 @@
+"""Common utility functions for NASweb integration."""
+
+from collections.abc import Mapping
+import logging
+from typing import Any, Optional
+
+from homeassistant.components.http import ApiConfig
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_HA_ADDRESS, PUSH_API_ENDPOINT
+from .coordinator import NotificationCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def initialize_notification_coordinator(
+    hass: HomeAssistant,
+) -> Optional[NotificationCoordinator]:
+    """Initialize and set up NotificationCoordinator instance."""
+    notifi_coordinator = NotificationCoordinator()
+    try:
+        _LOGGER.debug("Adding push notification handler")
+        hass.http.app.router.add_post(
+            PUSH_API_ENDPOINT, notifi_coordinator.handle_notification
+        )
+    except RuntimeError:
+        _LOGGER.error(
+            "Integration cannot register endpoint for local push. "
+            "Home Assistant restart should fix this issue"
+        )
+        return None
+    return notifi_coordinator
+
+
+def get_hass_address_from_entry(
+    hass: HomeAssistant, data: Mapping[str, Any]
+) -> Optional[str]:
+    """Return HA address for use in NASweb push api."""
```

Comment:
Is the intention here to find the local address for reaching the Home Assistant instance?

Reply:
During tests with HA within Docker container I had situation where hass.config.api.host returned container network interface (e.g. 10.0.3.16) when devices needed to connect with device's real network interface (e.g. 192.168.1.100). I assumed there would be simillar situation when HA is behind ssh tunneling so I added option to config flow as a workaround to pass address which devices should use to connect to my push api endpoint registered within HA.